src/main/scala/com/gradle/internal/CustomBuildScanEnhancements.scala

```diff
@@ -1,328 +1,369 @@
 package com.gradle.internal
 
 import com.gradle.develocity.agent.sbt.api.configuration.BuildScan
-import com.gradle.internal.CiUtils.*
-import com.gradle.internal.Utils.*
-import sbt.URL
+import com.gradle.develocity.agent.sbt.api.configuration.Server
+import com.gradle.internal.Utils.Env
+import java.util.Properties
+import java.net.URL
+import sbt.Logger
 
 /**
  * Adds a standard set of useful tags, links and custom values to all build scans published.
  */
-class CustomBuildScanEnhancements(serverConfig: ServerConfigTemp, scalaVersions: String) {
-
-  private val buildScan = new BuildScanTemp()
-
-  private val SYSTEM_PROP_IDEA_VENDOR_NAME = "idea.vendor.name"
-  private val SYSTEM_PROP_IDEA_VERSION = "idea.version"
-  private val SYSTEM_PROP_IDEA_MANAGED = "idea.managed"
-  private val SYSTEM_PROP_ECLIPSE_BUILD_ID = "eclipse.buildId"
-  private val SYSTEM_PROP_IDEA_SYNC_ACTIVE = "idea.sync.active"
+class CustomBuildScanEnhancements(serverConfig: Server, scalaVersions: Seq[String], logger: Logger)(implicit
+    env: Env
+) extends Transformer[BuildScan] {
+
+  override def transform(originBuildScan: BuildScan): BuildScan = {
+    val ops = Seq(
+      captureOs,
+      captureIde,
+      captureCiOrLocal,
+      captureCiMetadata,
+      captureGitMetadata,
+      (_: BuildScan).value("Scala versions", scalaVersions.mkString(","))
+    )
+    Function.chain(ops)(originBuildScan)
+  }
 
-  def withAdditionalData(originBuildScan: BuildScan): BuildScan = {
-    captureOs()
-    captureIde()
-    captureCiOrLocal()
-    captureCiMetadata()
-    captureGitMetadata()
+  private val captureOs = {
+    ifDefined(env.sysProperty[String]("os.name"))(_.tag(_))
+  }
 
-    buildScan.addValue("Scala versions", scalaVersions.mkString(","))
+  private val captureIde: BuildScan => BuildScan =
+    if (!CiUtils.isCi) identity
+    else {
+      val (ide, version) =
+        env
+          .sysProperty[String]("idea.vendor.name")
+          .filter(_ == "JetBrains")
+          .map(_ => ("IntelliJ IDEA", env.sysProperty[String]("idea.version")))
+          .orElse(env.sysProperty[String]("idea.version").map(v => ("IntelliJ IDEA", Some(v))))
+          .orElse(env.sysProperty[String]("idea.managed").map(_ => ("IntelliJ IDEA", None)))
+          .orElse(env.sysProperty[String]("eclipse.buildId").map(v => ("Eclipse", Some(v))))
+          .getOrElse(("Cmd Line", None))
+
+      val ops = Seq(
+        ifDefined(env.sysProperty[Unit]("idea.sync.active"))((bs, _) => bs.tag("IDE sync")),
```

Comment:
Is there a thing like IDE sync for sbt? Will this happen?

Reply:
There is, but I didn't find any evidence that IntelliJ or the Scala plugin for IntelliJ would set this particular system property. Maybe there's another one. I'll remove it in the meantime.